src/Sounder/Sounder.tsx

```diff
@@ -1,68 +1,72 @@
-import * as Tone from 'tone'
-import speaker from '../img/speaker.png'
-import React, {
-    ChangeEventHandler,
-    FC,
-    useEffect,
-    useMemo,
-    useState,
-} from "react"
+import * as Tone from "tone"
+import { FC, useState, startTransition } from "react"
 
-import { VolumeOffIcon, VolumeUpIcon } from '@heroicons/react/outline'
-import { Switch } from '@headlessui/react'
+import { VolumeUpIcon } from "@heroicons/react/outline"
+import { Switch } from "@headlessui/react"
 
-import { wait } from '@testing-library/user-event/dist/utils'
+import * as Note from "../model/note"
+import * as Scale from "../model/scales"
 
-const Sounder: FC = () => {
-    const synth = new Tone.Synth().toDestination();
-    const availableNotes = ["C4", "D4", "E4", "F4", "G4", "A4", "B4"]
-    const [long, setLength] = useState(false)
-    const [currentNote, setNote]=useState("C4")
+const synth = new Tone.Synth().toDestination()
 
-    const handleSound = (): void => {
-        handleNote()
-        synth.triggerAttackRelease(currentNote, handleLength(long))
-    }
+const Sounder: FC = () => {
+  const availableNotes: Note.Note[] = Scale.getScale("A", "major")
+  const [length, setLength] = useState(false)
+  const [currentNote, setCurrentNote] = useState<Note.Note>(
+    Note.initNote(availableNotes),
+  )
 
-    const handleLength = (longness: boolean): string => {
-        if (!longness) {
-            return "50n"
-        }
-        else {
-            return "8n"
-        }
-    }
+  const handleSound = (): void => {
+    const randomInteger = Math.floor(Math.random() * availableNotes.length)
+    const randomNote: Note.Note = availableNotes[randomInteger]
+    //handleNote()
+    synth.triggerAttackRelease(Note.getNote(randomNote), handleLength(length))
+    startTransition(() => {
+      setCurrentNote(randomNote)
+    })
+  }
 
-    const handleNote =() : void =>{
-        const random = Math.floor(Math.random() * availableNotes.length);
-        setNote(availableNotes[random])
+  const handleLength = (longness: boolean): string => {
```

Comment:
A few things not following conventions here:

1. Boolean variables are prefixed with `is`, so, while not as cool sounding, this should be be `isLong`

2. `handle` is usually reserved for callbacks for events or side effects, `handleOnClick` `handleReceivedResponse`, etc. as where this more of a business logic conditional based on some state. so I might use the name `determineLoudnessDuration` or something for readability.

In fact you might want to introduce a type for this concept as 'Loudness' seems to be a domain concept for you here. (the lack of which maybe why you have these code smells in the first place) e.g.

```typescript
type Loudness = "Loud" | "Quite"

const toDurationString = (loudness: Loudness): string => {
  switch (loudness) {
    case "Loud":
      return "50n"
    case "Quite"
      return "8n"
  }
}
```

Which may seem like a bit of boiler plate, but by naming the concept accurately in your type system and avoiding primitives, like `boolean`, you leave yourself open for extension and closed to conflation.

e.g. consider how you might introduce a new loudness value, "Medium", in the typed case, vs. the case where you are modeling the concept with `true` and `false` only.